### PR TITLE
chore: update deps

### PR DESCRIPTION
**Pull request by aschmahmann**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,13 +2,13 @@ module github.com/ipfs/go-ipfs
...
 	bazil.org/fuse v0.0.0-20200117225306-7b5117fecadc
-	contrib.go.opencensus.io/exporter/prometheus v0.3.0
+	contrib.go.opencensus.io/exporter/prometheus v0.4.0
 	github.com/blang/semver/v4 v4.0.0
 	github.com/cheggaaa/pb v1.0.29
 	github.com/coreos/go-systemd/v22 v22.3.2
 	github.com/dustin/go-humanize v1.0.0
 	github.com/elgris/jsondiff v0.0.0-20160530203242-765b5c24c302
-	github.com/fsnotify/fsnotify v1.4.9
+	github.com/fsnotify/fsnotify v1.5.1
```
Comment (aschmahmann) [inline go.mod:11]: https://github.com/fsnotify/fsnotify/releases

While v1.5.0 was a buggy release that got retracted v1.5.1 seems to basically just fix an unsafe pointer conversion (https://github.com/fsnotify/fsnotify/pull/325) and update x/sys

https://github.com/fsnotify/fsnotify/compare/v1.4.9...v1.5.1

File: go.mod [hunk 2]
```diff
@@ -90,22 +90,22 @@ require (
 	github.com/libp2p/go-ws-transport v0.5.0
 	github.com/miekg/dns v1.1.43
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/multiformats/go-multiaddr v0.4.0
+	github.com/multiformats/go-multiaddr v0.4.1
```
Comment (aschmahmann) [inline go.mod:93]: plaintextv2 code point added https://github.com/multiformats/go-multiaddr/releases/tag/v0.4.1

File: go.mod [hunk 3]
```diff
@@ -26,7 +26,7 @@ require (
 	github.com/ipfs/go-fetcher v1.6.1
 	github.com/ipfs/go-filestore v0.1.0
 	github.com/ipfs/go-fs-lock v0.0.7
-	github.com/ipfs/go-graphsync v0.10.6-0.20211119194020-99752d2dc1cc
+	github.com/ipfs/go-graphsync v0.11.0
```
Comment (aschmahmann) [inline go.mod:29]: go-datastore contexts merged + released

File: go.mod [hunk 4]
```diff
@@ -90,22 +90,22 @@ require (
...
+	github.com/prometheus/statsd_exporter v0.22.4 // indirect
 	github.com/stretchr/testify v1.7.0
 	github.com/syndtr/goleveldb v1.0.0
 	github.com/whyrusleeping/go-sysinfo v0.0.0-20190219211824-4a357d4b90b1
 	github.com/whyrusleeping/multiaddr-filter v0.0.0-20160516205228-e903e4adabd7
 	go.opencensus.io v0.23.0
-	go.uber.org/fx v1.13.1
-	go.uber.org/zap v1.19.0
+	go.uber.org/fx v1.15.0
+	go.uber.org/zap v1.19.1
```
Comment (aschmahmann) [inline go.mod:108]: some small JSON encoding fixes that likely don't effect us https://github.com/uber-go/zap/releases/tag/v1.19.1

File: go.mod [hunk 5]
```diff
@@ -90,22 +90,22 @@ require (
...
-	github.com/prometheus/statsd_exporter v0.21.0 // indirect
+	github.com/prometheus/statsd_exporter v0.22.4 // indirect
 	github.com/stretchr/testify v1.7.0
 	github.com/syndtr/goleveldb v1.0.0
 	github.com/whyrusleeping/go-sysinfo v0.0.0-20190219211824-4a357d4b90b1
 	github.com/whyrusleeping/multiaddr-filter v0.0.0-20160516205228-e903e4adabd7
 	go.opencensus.io v0.23.0
-	go.uber.org/fx v1.13.1
-	go.uber.org/zap v1.19.0
+	go.uber.org/fx v1.15.0
```
Comment (aschmahmann) [inline go.mod:107]: It looks like there have been a number of changes to fx in the last few releases (i.e. between Aug 2020 and Nov 2021).

v0.15.0 was released a few weeks ago and has some interesting features available (e.g. `fx.Annotate` that could allow us to not need as many `fx.In/Out` wrapper structs). If we're not comfortable upgrading yet we could stay on v1.14.2 as well.

File: go.mod [hunk 6]
```diff
@@ -90,22 +90,22 @@ require (
...
+	github.com/multiformats/go-multiaddr v0.4.1
 	github.com/multiformats/go-multiaddr-dns v0.3.1
 	github.com/multiformats/go-multibase v0.0.3
 	github.com/multiformats/go-multicodec v0.3.0
 	github.com/multiformats/go-multihash v0.1.0
 	github.com/opentracing/opentracing-go v1.2.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.11.0
-	github.com/prometheus/statsd_exporter v0.21.0 // indirect
+	github.com/prometheus/statsd_exporter v0.22.4 // indirect
```
Comment (aschmahmann) [inline go.mod:101]: I don't know if we need this to be a manual/indirect import anymore (the latest is v0.22.4). @marten-seemann why was this done originally?

Comment (aschmahmann) [reply go.mod]: Not bumping this explicitly, and this indirect import is covered by importing `contrib.go.opencensus.io/exporter/prometheus v0.4.0`

File: go.mod (hunk 2 above)
Comment (marten-seemann) [reply go.mod:93]: We probably don’t need to do this, unless this is pulled in by some other dependency update. On the other, it won’t hurt either.

File: go.mod (hunk 6 above)
Comment (marten-seemann) [reply go.mod]: I’m not sure, I assume it got pulled in by either go-libp2p-quic-transport or go-tcp-transport.

**Review by lidel — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (aschmahmann) [reply go.mod:94]: ya, we don't need it but figure for dependencies we control we should probably try and run with the latest so we can detect any issues

**Pull request merged**